src/document.c

```diff
@@ -175,6 +175,11 @@ RSAddDocumentCtx *NewAddDocumentCtx(IndexSpec *sp, Document *b, QueryError *stat
   aCtx->next = NULL;
   aCtx->specFlags = sp->flags;
   aCtx->indexer = sp->indexer;
+  aCtx->spec = sp;
+  if (aCtx->specFlags & Index_Async) {
+    aCtx->specName = rm_strdup(sp->name);
```

Comment:
aCtx is allocated from a pool so it might be a good idea to reuse the memory here (allocate once and reallocate if a larger name arrives), what do you think?